src/App.tsx

```diff
@@ -15,47 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  LENGHT,
+  ALPABETICALLY,
+  NONE,
+}
+
+type OrderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+function orderGoods(goods: string[],
+  { sortType, isReversed }: OrderOptions) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((goodA: string, goodB: string) => {
+    switch (sortType) {
+      case SortType.ALPABETICALLY:
+        return goodA.localeCompare(goodB);
+      case SortType.LENGHT:
+        return goodA.length - goodB.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = orderGoods(goodsFromServer, { sortType, isReversed });
+
+  const sortAlphabetically = () => {
+    setSortType(SortType.ALPABETICALLY);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.LENGHT);
+  };
+
+  const sortReverse = () => {
+    setIsReversed(current => !current);
+  };
+
+  const sortReset = () => {
+    setSortType(SortType.NONE);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info', {
+            'is-light': sortType !== SortType.ALPABETICALLY,
+          })}
+          onClick={sortAlphabetically}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success', {
+            'is-light': sortType !== SortType.LENGHT,
+          })}
+          onClick={sortByLength}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={sortReverse}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortType !== SortType.NONE || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={sortReset}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {visibleGoods.map(good => (
+            <li data-cy="Good">{good}</li>
```

Comment:
Don't forget about key attribute 
![image](https://github.com/mate-academy/react_list-of-goods-hooks/assets/96409563/4625fbac-f955-4a5d-a57c-7fe9b9b429b7)